examples/text-generation/run_generation.py

```diff
@@ -52,9 +52,15 @@ def generate(model, tokenizer, prompts, length, temperature):
         default="One of my fondest memory is",
         help="The prompts to use for generation, using | as separator.",
     )
+    parser.add_argument(
+        "--batch_size",
+        type=int,
+        default=None,
+        help="The batch size (optional). If not specified it wil be based on the number of prompts.",
+    )
     parser.add_argument("--length", type=int, default=128, help="The number of tokens in the generated sequences.")
     parser.add_argument(
-        "--num_cores", type=int, default=2, help="The number of cores on which the model should be split."
+        "--num_cores", type=int, default=1, help="The number of cores on which the model should be split."
```

Comment:
Because I realized that for small models and small batch sizes it is much faster to use a single core than 2.
For instance for GPT2, even with batch size 16 it is still better to use a single core. For a batch size of 128 however, it will be faster if you use two cores.